app/src/main/java/com/space/quizapp/presentation/ui/home/QuizHomeFragment.kt

```diff
@@ -22,13 +25,37 @@ class QuizHomeFragment : QuizBaseFragment<QuizHomeViewModel>() {
         navigate()
     }
 
-    private fun showGpaScore(){
-        //TODO - GET CURRENT AUTH USERNAME AND PASS IT TO THE FUNCTION (DATASTORE STATE HANDLER)
-        viewModel.loadUserPoints("lukaadmin")
-        viewModel.userPoints.observe(this){
-            binding.gpaTextView.text = "GPA - ${it.toString()}"
+    private fun showGpaScore() {
+        viewModel.updateSession()
+
+        /**
+         * Observe the session and load the user points
+         */
+        viewModel.session.observe(viewLifecycleOwner) { session ->
+            session?.let {
+                binding.helloUserTextView.text = buildString {
+                    append(getString(R.string.hello_user))
+                    append(it)
+                }
+                viewModel.loadUserPoints(it)
```

Comment:
what is buildString for

Reply:
![image](https://github.com/iTarley/QuizApp-SpaceIntern/assets/95241918/0504834c-32db-409c-b2e5-e4047dad88bd)